satellite/console/consoleweb/server.go

```diff
@@ -192,6 +217,323 @@ func (server *Server) appHandler(w http.ResponseWriter, r *http.Request) {
+		Password string `json:"password"`
+	}
+
+	var tokenRequest tokenRequestModel
+	err = json.NewDecoder(r.Body).Decode(&tokenRequest)
+	if err != nil {
+		server.serveJsonError(w, 400, err)
+	}
+
+	token, err := server.service.Token(ctx, tokenRequest.Email, tokenRequest.Password)
+	if err != nil {
+		server.serveJsonError(w, 404, err)
+		return
+	}
+
+	err = json.NewEncoder(w).Encode(token)
+	if err != nil {
+		server.serveJsonError(w, 500, err)
+		server.log.Debug("Error serializing response: " + err.Error())
+	}
+}
+
+// changeAccountPasswordRequestHandler updates password for a given user
+func (server *Server) changeAccountPasswordRequestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var err error
+	defer mon.Task()(&ctx)(&err)
+
+	_, err = console.GetAuth(ctx)
+	if err != nil {
+		server.serveJsonError(w, 401, err)
+		return
+	}
+
+	type ChangePasswordRequestModel struct {
+		Password    string `json:"password"`
+		NewPassword string `json:"newPassword"`
+	}
+
+	var passwordChange ChangePasswordRequestModel
+	err = json.NewDecoder(r.Body).Decode(&passwordChange)
+	if err != nil {
+		server.serveJsonError(w, 400, err)
+		return
+	}
+
+	err = server.service.ChangePassword(ctx, passwordChange.Password, passwordChange.NewPassword)
+	if err != nil {
+		server.serveJsonError(w, 404, err)
+		return
+	}
+}
+
+// createNewUserRequestHandler gets password hash value and creates new inactive User
+func (server *Server) createNewUserRequestHandler(w http.ResponseWriter, r *http.Request) {
+	ctx := r.Context()
+	var err error
+	defer mon.Task()(&ctx)(&err)
+
+	type createUserRequestModel struct {
+		console.CreateUser
+		Secret         string `json:"secret"`
+		ReferrerUserID string `json:"referrerUserId"`
+	}
+
+	var model createUserRequestModel
+	err = json.NewDecoder(r.Body).Decode(&model)
+	if err != nil {
+		server.serveJsonError(w, 400, err)
+		return
+	}
+
+	secret, err := console.RegistrationSecretFromBase64(model.Secret)
+	if err != nil {
+		log.Error("register: failed to create account",
+			zap.Error(err))
+		log.Debug("register: ", zap.String("rawSecret", model.Secret))
+		server.serveJsonError(w, 400, err)
+
+		return
+	}
+
+	user, err := server.service.CreateUser(ctx, model.CreateUser, secret, model.ReferrerUserID)
+	if err != nil {
+		log.Error("register: failed to create account",
+			zap.Error(err))
+		log.Debug("register: ", zap.String("rawSecret", model.Secret))
+		server.serveJsonError(w, 400, err)
+
+		return
+	}
+
+	token, err := server.service.GenerateActivationToken(ctx, user.ID, user.Email)
+	if err != nil {
+		log.Error("register: failed to generate activation token",
+			zap.Stringer("id", user.ID),
+			zap.String("email", user.Email),
+			zap.Error(err))
+		server.serveJsonError(w, 400, err)
+
+		return
+	}
+
+	rootObject, ok := ctx.Value("rootObject").(RootObject)
+	if !ok {
+		server.log.Error("root object is not set")
+		return
+	}
+	link := rootObject.Origin + rootObject.ActivationPath + token
```

Comment:
This method is on the same level with `server`, so it has access to `config` and routes

Reply:
Removed